### PR TITLE
add MySQL 8.4 in `google_sql_database_instance` documentation

### PR DESCRIPTION
**Pull request by translucens**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/20579

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
sql: added MySQL 8.4 in `google_sql_database_instance` documentation
```

### CODE REVIEW

File: mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown [hunk 1]
```diff
@@ -239,7 +239,7 @@ The following arguments are supported:
 
 * `database_version` - (Required) The MySQL, PostgreSQL or
 SQL Server version to use. Supported values include `MYSQL_5_6`,
-`MYSQL_5_7`, `MYSQL_8_0`, `POSTGRES_9_6`,`POSTGRES_10`, `POSTGRES_11`,
+`MYSQL_5_7`, `MYSQL_8_0`, `MYSQL_8_4`, `POSTGRES_9_6`,`POSTGRES_10`, `POSTGRES_11`,
```
Comment (zli82016) [inline mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown:242]: The `MYSQL_8_4` is only added to the documentation. Does it need to be added to somewhere else to be supported?

Comment (translucens) [reply mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown:242]: Updated PR title

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**